### PR TITLE
added dockerfile for deploying app on container

### PR DESCRIPTION
**Pull request by captain-Akshay**
Issue #378

This commit includes a Dockerfile and a Docker Compose file to enhance the deployment of the app on a local setup.

However, there is still an issue with containerizing Next.js version 13.4.20-canary.31. To learn more and if you encounter a related issue, please visit https://github.com/vercel/next.js/issues?q=docker+.

I've also added a Docker Compose file with a MySQL server for communication between the containers and made a few minor changes to improve the development experience.

### CODE REVIEW

**Review by Ashutosh-Malve — APPROVED** [review]

**Review by Ashutosh-Malve — APPROVED** [review]

**Comment by steven-tey** [comment]
@captain-Akshay we recently refactored the codebase to a [monorepo via Turborepo](https://dub.co/blog/turborepo-migration) – can we refactor this PR please?

**Comment by captain-Akshay** [comment]
> @captain-Akshay we recently refactored the codebase to a [monorepo via Turborepo](https://dub.co/blog/turborepo-migration) – can we refactor this PR please?

sure let me take a look

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM node:18-alpine
+
+# Check https://github.com/nodejs/docker-node/tree/b4117f9333da4138b03a546ec926ef50a31506c3#nodealpine to understand why libc6-compat might be needed.
+RUN apk add --no-cache libc6-compat
+WORKDIR /app
+
+COPY package.json yarn.lock ./
+
+RUN yarn
```
Comment (mcnaveen) [inline Dockerfile:9]: Running `yarn` works. But I suggest using `yarn install` instead. Much more clear for beginners

**Comment by skushagra9** [comment]
If this issue is not solved ,please assign it to me

**Comment by captain-Akshay** [comment]
apologies for the delay i was busy with some intern work i will finish it up by today!

File: .env.example
Comment (steven-tey) [inline .env.example:1]: don't think this change is required

**Review by steven-tey — CHANGES_REQUESTED** [review]

**Comment by Nasir1999** [comment]
@captain-Akshay any update? ☺

**Comment by juliomfreitas** [comment]
Finishing this PR would be great for the community. thank you all guys!!!

**Comment by captain-Akshay** [comment]
Sorry I'm currently busy with something else, you can gladly take this one

**Comment by fredygerman** [comment]
i see the issue has been closed, has it been merged and is it done?

**Comment by gabriel-vanca** [comment]
Why was this issue closed? It doesn't seem to have been solved.

**Comment by jsilvermist** [comment]
I see this PR is still referenced on the [Self-hosting documentation](https://dub.co/docs/self-hosting#step-3%3A-set-up-upstash-redis-database:~:text=Also%2C%20Docker%20is%20currently%20not%20supported%2C%20but%20we%20have%20a%20few%20open%20issues%20and%20PRs%20for%20it.), and would really like this as it's the ideal way to deploy this on my setup, has this been abandoned permanently?

**Pull request closed without merging**